financial-connections/src/main/java/com/stripe/android/financialconnections/features/consent/ConsentViewModel.kt

```diff
@@ -52,33 +52,36 @@ internal class ConsentViewModel @Inject constructor(
                 shouldShowMerchantLogos = shouldShowMerchantLogos,
                 merchantLogos = sync.visual.merchantLogos
             )
-        }.execute { copy(consent = it) }
-    }
-
-    private fun logErrors() {
-        onAsync(
-            ConsentState::consent,
+        }.execute(
+            reducer = { copy(consent = it) },
             onSuccess = { eventTracker.track(PaneLoaded(Pane.CONSENT)) },
-            onFail = { logger.error("Error retrieving consent content", it) }
+            onFail = {
```

Comment:
Not sure how to change it, but having all three in the same method feels a little bloated.

Reply:
we can introduce the concept of `onEach` (observing asyncs) on the base viewmodel as well, although I also like the contextual side effects (but I'm with you on the "this is too much" thinking.